homeassistant/components/climate/fritzhome.py

```diff
@@ -0,0 +1,132 @@
+
+    add_devices(devices)
+
+
+class FritzhomeThermostat(ClimateDevice):
+    """The thermostat class for Fritzhome thermostates."""
+
+    def __init__(self, device):
+        """Initialize the thermostat."""
+        self._device = device
+        self._current_temperature = self._device.actual_temperature
+        self._target_temperature = self._device.target_temperature
+        self._comfort_temperature = self._device.comfort_temperature
+        self._eco_temperature = self._device.eco_temperature
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def available(self):
+        """Return if thermostat is available."""
+        return self._device.present
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._device.name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement that is used."""
+        return TEMP_CELSIUS
+
+    @property
+    def precision(self):
+        """Return precision 0.5."""
+        return PRECISION_HALVES
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if ATTR_OPERATION_MODE in kwargs:
+            operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+            self.set_operation_mode(operation_mode)
+        elif ATTR_TEMPERATURE in kwargs:
+            temperature = kwargs.get(ATTR_TEMPERATURE)
+            self._device.set_target_temperature(temperature)
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        if self._target_temperature == self._comfort_temperature:
+            return STATE_HEAT
+        elif self._target_temperature == self._eco_temperature:
+            return STATE_ECO
+        return STATE_MANUAL
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return OPERATION_LIST
+
+    def set_operation_mode(self, operation_mode):
+        """Set new operation mode."""
+        if operation_mode == STATE_HEAT:
+            self.set_temperature(temperature=self._comfort_temperature)
+        elif operation_mode == STATE_ECO:
+            self.set_temperature(temperature=self._eco_temperature)
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return MIN_TEMPERATURE
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return MAX_TEMPERATURE
+
+    def update(self):
+        """Update the data from the thermostat."""
+        self._device.update()
```

Comment:
To reconnect, add `try... except` around this call, and call the reconnect method in the `except` block.

Reply:
I found a way to do a reconnect. Just added the code and pushed.